iwyu_ast_util.cc

```diff
@@ -888,19 +887,23 @@ static const Type* GetSugaredTypeOf(const Expr* expr) {
   return v.sugared.getTypePtr();
 }
 
-map<const Type*, const Type*> GetTplTypeResugarMapForFunction(
-    const FunctionDecl* decl, const Expr* calling_expr) {
-  map<const Type*, const Type*> retval;
+TemplateInstantiationData GetTplInstDataForFunction(
+    const FunctionDecl* decl, const Expr* calling_expr,
+    std::function<set<const clang::Type*>(const clang::Type*)>
```

Comment:
`clang::Type` already seems to be imported into the current namespace, so you should be able to simplify this signature just using `Type`